### PR TITLE
BI-1307 Create rspec test for 'artifacts' function

### PR DESCRIPTION
**Pull request by vovuong**
1. it 'result artifacts.error returns if the returned build_id is empty' do
2. it 'No artifact was found if have_build_number = false and detail is empty' do
3. it 'No artifact was found if have_build_number = true and detail is empty' do
4. it 'result returns correctly if have_build_number = false and detail is not empty' do
5. it 'result returns correctly if have_build_number = true and detail is not empty' do

### CODE REVIEW

**Comment by thanhhoang92** [comment]
In my opinion, this function should have only 4 test cases that corresponds to 4 cases of conditions (if-else). You only focus workflow of the main function and please don't check conditions of used helper functions .

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 1]
```diff
@@ -197,4 +197,94 @@
...
+                            and_return([valid_build_number, real_build_number])
+      allow_any_instance_of(Lita::Handlers::Teamcity).to receive(:artifacts_by_build_id).\
+                            and_return(artifacts_result_without_data)
+      send_command("artifacts #{valid_build_name} #{valid_build_number}")
+      expect(replies.last).to eq(expected_output)
+    end
+
+    it 'result returns correctly if have_build_number = false and detail is not empty' do
+      expected_output = "Here are the artifacts of <#{registry.config.handlers.teamcity.site}/" +
+                  "viewLog.html?buildId=#{valid_build_number}&buildTypeId=#{valid_build_name}&tab=" +
```
Comment (thanhhoang92) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: please align

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 2]
```diff
@@ -197,4 +197,68 @@
...
+      expected_output = "Cannot find build #{invalid_build_name} ()!"
+      allow_any_instance_of(Lita::Handlers::Teamcity).to receive(:id_build).and_return(["", ""])
+      send_command("artifacts #{invalid_build_name}")
+      expect(replies.last).to eq(expected_output)
+    end
+
+    it 'No artifact was found if have_build_number = false and detail is empty' do
+      expected_output = "No artifact was found in the given build number ()!"
+      allow_any_instance_of(Lita::Handlers::Teamcity).to receive(:id_build).\
+                            and_return([valid_build_number, real_build_number])
```
Comment (nguyentranlogigear) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: Because allow_any_instance_of use many times. So we can put it in Before do

**Comment by nguyentranlogigear** [comment]
LGTM

**Pull request merged**